src/main/java/io/aiven/kafka/connect/http/sender/HttpSender.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2019 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class HttpSender {
+
+    private static final Logger log = LoggerFactory.getLogger(HttpSender.class);
+
+    protected static final String HEADER_AUTHORIZATION = "Authorization";
+
+    protected static final String HEADER_CONTENT_TYPE = "Content-Type";
+
+    private static final Duration HTTP_TIMEOUT = Duration.ofSeconds(30);
+
+    private final HttpClient httpClient;
+
+    protected final HttpRequest.Builder requestTemplate;
+
+    protected final HttpSinkConfig config;
+
+    protected interface OnResponseHandler {
+
+        void onResponse(final HttpResponse<String> response) throws IOException;
+
+    }
+
+    protected final OnResponseHandler onHttpErrorResponseHandler = response -> {
+        if (response.statusCode() >= 400) {
+            log.warn("Got unexpected HTTP status code: {}", response.statusCode());
```

Comment:
Should this be under if (response.statusCode < 200 || response.statusCode > 299) ???